src/watchdog.rs

```diff
@@ -0,0 +1,103 @@
+//! API for the IWDG
+//! You can activate the Watchdog by callind `start` or setting appropriate
+//! device option bit when programming.
+//!
+//! After activating it, you'll have to regularly `feed` the watchdog in at
+//! least the interval specified in the start method, otherwise the
+//! Microcontroller will be reset.
+//!
+//! This is usefull if you fear that your program may get stuck. Then it won't
+//! feed the Watchdog anymore, it will be reset and thus functional again.
+//!
+//! Attention:
+//!
+//! The IWDG runs on a separate 40kHz low-accuracy Clock. You may want to
+//! include some buffer in your interval.
+//!
+//! # Example
+//! ``` no_run
+//! use stm32f0xx_hal as hal;
+//!
+//! use crate::hal::stm32;
+//! use crate::hal::prelude::*;
+//! use crate::hal:watchdog::Watchdog;
+//! use crate::hal:time::Hertz;
+//!
+//! let mut p = stm32::Peripherals::take().unwrap();
+//!
+//! let mut iwdg = Watchdog::new(p.iwdg);
+//! iwdg.start(Hertz(100));
+//! loop {}
+//! // Whoops, got stuck, the watchdog issues a reset after 10 ms
+//! iwdg.feed();
+//! ```
+use embedded_hal::watchdog;
+
+use crate::stm32::IWDG;
+use crate::time::Hertz;
+
+pub struct Watchdog {
```

Comment:
Doc tags on these structs would be nice, just so they get descriptions in lists in the docs.